R/utils.R

```diff
@@ -100,6 +100,36 @@ paste_lines = function(lines) {
 Abort = function(msg, ...) rlang::abort(break_str(msg, init = "Error: "), ...)
 Warn = function(msg, ...) rlang::warn(break_str(msg, init = "Warning: "), ...)
 
+#' Check that a sliding computation function takes enough args
+#'
+#' @param f Function; specifies a computation to slide over an `epi_df` or
+#'  `epi_archive` in `epi_slide` or `epix_slide`.
+#'
+#' @noRd
+check_sufficient_f_args <- function(f) {
+  n_mandatory_f_args <- 2
+  arg_names = names(formals(args(f)))
+  if ("..." %in% arg_names) {
+    # Keep all arg names before `...`
+    dots_i <- which(arg_names == "...")
+    arg_names <- arg_names[seq_len(dots_i - 1)]
+
+    if (length(arg_names) < n_mandatory_f_args) {
```

Comment:
issue: While this should catch the `slice_head` issue, I don't think this will catch something like trying to use `lm` and having some of its parameters-that-have-defaults filled with the group key / ref_time_value.  This could probably be checked by seeing if any of the first `n_mandatory_f_args` `formals(args(f))` are `identical` to `quote(expr =)` or `rlang::expr()` --- see https://rlang.r-lib.org/reference/missing_arg.html.

Reply:
Added in https://github.com/cmu-delphi/epiprocess/commit/d5466d38db3a149fe2a990e6f91b1561836abab0

I used `rlang::is_missing` since it [already contains the `identical(x, quote(expr =)` check](https://github.com/r-lib/rlang/blob/84bd450/R/arg.R#L463-L465) you mentioned.